azure-pipelines.yml

```diff
@@ -0,0 +1,109 @@
+variables:
+  PreferredToolArchitecture: x64
+      
+jobs:
+- job: Windows
+  timeoutInMinutes: 0
+  pool:
+    vmImage: 'vs2017-win2016'
+  strategy:
+    matrix:
+      ITK4:
+        itk.version: 'release-4.13'
+      ITK5:
+        itk.version: 'e018653622038ca7651d21769acfec539f01344d' # master branch as of 22-01-2019
+  steps:
+  - script: |
+      git clone --branch $(itk.version) https://github.com/InsightSoftwareConsortium/ITK $(Agent.HomeDirectory)\ITK
+      mkdir $(Agent.BuildDirectory)\ITK
+    displayName: Clone ITK
+  - task: CMake@1
+    displayName: 'CMake Generate ITK'
+    inputs:
+      cmakeArgs: -G "Visual Studio 15 2017 Win64" -T host=x64 -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF $(Agent.HomeDirectory)\ITK
+      workingDirectory: $(Agent.BuildDirectory)\ITK
+  - task: CMake@1
+    displayName: 'CMake Build ITK'
+    inputs:
+      cmakeArgs: --build . --config Release -j 2
+      workingDirectory: $(Agent.BuildDirectory)\ITK
+  - task: CMake@1
+    displayName: 'CMake Generate Elastix'
+    inputs:
+      cmakeArgs: -G "Visual Studio 15 2017 Win64" -T host=x64 -DITK_DIR=$(Agent.BuildDirectory)\ITK ..
+  - task: CMake@1
+    displayName: 'CMake Build Elastix'
+    inputs:
+      cmakeArgs: --build . --config Release -j 2
+  - script: ctest -VV -j 2
+    displayName: 'CTest Elastix'
+- job: Linux
+  timeoutInMinutes: 0
+  pool:
+    vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
+      ITK4:
+        itk.version: 'release-4.13'
+      ITK5:
+        itk.version: 'e018653622038ca7651d21769acfec539f01344d' # master branch as of 22-01-2019
+  steps:
+  - script: |
+      git clone --branch $(itk.version) https://github.com/InsightSoftwareConsortium/ITK $(Agent.HomeDirectory)/ITK
+      mkdir $(Agent.BuildDirectory)/ITK
+    displayName: Clone ITK
+  - task: CMake@1
+    displayName: 'CMake Generate ITK'
+    inputs:
+      cmakeArgs: -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF $(Agent.HomeDirectory)/ITK
+      workingDirectory: $(Agent.BuildDirectory)/ITK
+  - task: CMake@1
+    displayName: 'CMake Build ITK'
+    inputs:
+      cmakeArgs: --build . --config Release -j 2
+      workingDirectory: $(Agent.BuildDirectory)/ITK
+  - task: CMake@1
+    displayName: 'CMake Generate Elastix'
+    inputs:
+      cmakeArgs: -DITK_DIR=$(Agent.BuildDirectory)/ITK ..
+  - task: CMake@1
+    displayName: 'CMake Build Elastix'
+    inputs:
+      cmakeArgs: --build . --config Release -j 2
+  - script: ctest -VV -j 2
+    displayName: 'CTest Elastix'
+- job: macOS
+  timeoutInMinutes: 0
+  pool:
+    vmImage: 'macOS-10.13'
+  strategy:
+    matrix:
+      ITK4:
+        itk.version: 'release-4.13'
+      ITK5:
+        itk.version: 'e018653622038ca7651d21769acfec539f01344d' # master branch as of 22-01-2019
```

Comment:
Would it be possible "share" this git-hash with the other sections, `job: Windows` and `job: Linux`? (No show-stopper.)

Reply:
Yes we can make a variable at the top (like "PreferredToolArchitecture") and reference this variable in all sections. Would like to give it a go?

Reply:
If azure is still running, I would first like to see it succeed. A "git hash" variable would certainly be nice, but it's no show-stopper to me, for the initial version.

Reply:
So far the failures seems to be related to the multi-threader changes. You have fixed this in another branch, right? If so, we can see on the pr-merge build if the issues are fully resolved before merging this PR.

Reply:
Oh, I see these changes were already merged. I can't figure out why some ITKv4 builds succeed while others don't, and some ITKv5 builds succeed and others don't. Can you take a look at the error messages azure?

Reply:
Indeed I did merge #104 to develop, this morning. I had it open for review fo 23 hours, and I thought it wasn't necessary to wait any longer. The commits can easily be reverted later, if necessary. So the develop branch should now compile for both ITK4 and ITK5.

Reply:
@kaspermarstal Linux ITKv5 says at https://dev.azure.com/kaspermarstal/Elastix/_build/results?buildId=127:

    Common/ImageSamplers/itkImageToVectorContainerFilter.hxx:278:35: error: ‘MultiThreader’ has not been declared

https://github.com/SuperElastix/elastix/blob/develop/Common/ImageSamplers/itkImageToVectorContainerFilter.hxx#L278	
	
Can you please try to fix it? It should be sufficient to just add `#include "itkMultiThreader.h"` to the elastix file "itkImageToVectorContainerFilter.hxx".

Reply:
Done, more progress, and we now get this: 

```
/home/vsts/work/1/s/Components/Metrics/AdvancedKappaStatistic/itkAdvancedKappaStatisticImageToImageMetric.hxx:673:86: error: there are no arguments to ‘vcl_ceil’ that depend on a template parameter, so a declaration of ‘vcl_ceil’ must be available [-fpermissive]
vcl_ceil( static_cast< double >( numPar ) / static_cast< double >( nrOfThreads ) ) );
```

Have you seen this before?